src/reducers/Theme.ts

```diff
@@ -0,0 +1,23 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+
+const initialState = {
+  mode: 'light',
+};
+
+const theme = createSlice({
+  name: 'theme',
+  initialState,
+  reducers: {
+    updateThemeMode(
+      state,
+      action: PayloadAction<{
+        mode: string;
+      }>,
```

Comment:
You can use a simpler payload:
```suggestion
      action: PayloadAction<string>,
```
and then
```js
    dispatch(
      updateThemeMode(storedMode === 'light' ? 'dark' : 'light'),
    );
```